sdk/storage/azure-storage-common/src/xml_wrapper.cpp

```diff
@@ -392,19 +396,29 @@ namespace Azure { namespace Storage { namespace _internal {
 
 #else
 
-  struct XmlGlobalInitializer final
+  void XmlGlobalInitialize()
   {
-    XmlGlobalInitializer() { xmlInitParser(); }
-    ~XmlGlobalInitializer() { xmlCleanupParser(); }
-  };
+    static std::once_flag flag;
+    std::call_once(flag, [] { xmlInitParser(); });
+  }
 
-  static void XmlGlobalInitialize() { static XmlGlobalInitializer globalInitializer; }
+  void XmlGlobalDeinitialize()
+  {
+    static std::once_flag flag;
+    std::call_once(flag, [] { xmlCleanupParser(); });
+  }
 
-  struct XmlReaderContext
+  struct XmlReader::XmlReaderContext
   {
-    xmlTextReaderPtr reader = nullptr;
+    using XmlTextReaderPtr = std::unique_ptr<xmlTextReader, decltype(&xmlFreeTextReader)>;
```

Comment:
Hi @rschu1ze @LarryOsterman , I have difficulty understanding this, can you help me here?

What does the 2nd template argument do? You specified a custom deleter at Line 434 when instantiating `XmlTextReaderPtr`, why do we need this 2nd template argument?

Reply:
The code is based on https://stackoverflow.com/a/19054467/22422288, the second template argument specifies the signature of the deleter.